File: snippets/fsharp/System/String/Concat/concat3.fs
module concat3

// <Snippet4>
// This example uses the F# Seq.filter function instead of Linq.
open System

type Animal =
  { Kind: string
    Order: string }
    override this.ToString() =
       this.Kind

let animals = ResizeArray()
animals.Add { Kind = "Squirrel"; Order = "Rodent" }
animals.Add { Kind = "Gray Wolf"; Order = "Carnivora" }
animals.Add { Kind = "Capybara"; Order = "Rodent" }

Seq.filter (fun animal -> animal.Order = "Rodent")
|> String.Concat
|> printfn "%s"
// The example displays the following output:
//      SquirrelCapybara
// </Snippet4>

File: snippets/csharp/System/String/Concat/concat3.cs
// <Snippet4>
using System;
using System.Collections.Generic;
using System.Linq;

public class Animal
{
    public string Kind;
    public string Order;

    public Animal(string kind, string order)
    {
        this.Kind = kind;
        this.Order = order;
    }

    public override string ToString() => this.Kind;
}

public class ConcatAnimalsExample
{
    public static void Run()
    {
        List<Animal> animals = new()
        {
            new Animal("Squirrel", "Rodent"),
            new Animal("Gray Wolf", "Carnivora"),
            new Animal("Capybara", "Rodent")
        };
        string output = string.Concat(animals.Where(animal =>
                        (animal.Order == "Rodent")));
        Console.WriteLine(output);
    }
}
// The example displays the following output:
//      SquirrelCapybara
// </Snippet4>
